### DISCUSSION TITLE
Resources outlining the best way to alter pytest terminal output

### DISCUSSION THREAD

**Discussion by harryjulian** [pytest-dev/pytest#11526] [General]
I'm currently writing a pytest plugin that will give users the ability to to accept/reject saving an artifact from a test, after the tests have ran (I'll ideally print the test code and some of it's output, and then give a [Y/n] response for the user). I want to add this after the regular pytest output in stdout, aiming to keep to the current styling. Are there any resources that might help me work out the best way to do it? It it just a matter of copying the formatting and running a hook after the tests have finished?

**Comment by nicoddemus** [comment]
I don't think there are specific guide lines for that.

Btw in case you don't know, the [pytest_terminal_summary](https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_terminal_summary) hook can be used for that.
